packages/webview_flutter/webview_flutter/lib/src/webview.dart

```diff
@@ -94,6 +96,7 @@ class WebView extends StatefulWidget {
     this.initialMediaPlaybackPolicy =
         AutoMediaPlaybackPolicy.require_user_action_for_all_media_types,
     this.allowsInlineMediaPlayback = false,
+    this.limitsNavigationsToAppBoundDomains = false,
```

Comment:
Adding platform-specific arguments to this list is not a direction I want to take if we can possibly avoid it; it could get out of hand very quickly.

@bparrishMines @mvanbeusekom You're both more familiar than I am with the overall API of this plugin; do either of you have thoughts on how to introduce platform-specific extensions here?

I'm wondering if it would make sense to add a creation parameters class that's created by a factory, implemented by the implementing packages, so that they could return subclasses of the settings object that have extensions. Then clients could do platform-specific casting to access settings like this one, similar to the IAP extensions.

Reply:
In the long term, I believe we could use casting to add cleaner support for very specific platform features like this. For example, the API would look something like:

```dart
class MyWidget extends StatelessElement {
  MyWidget(StatelessWidget widget) : super(widget);

  Future<void> additionalIosSetup(IosWebViewController controller) {
    controller.configuration.setLimitsNavigationsToAppBoundDomains(false);
  }

  Future<void> additionalAndroidSetup(AndroidWebViewController controller) {
    ...
  }

  @override
  Widget build() {
    return WebView(
      onWebViewCreated: (WebViewController controller) {
        if (Platform.isIOS) {
          additionalIosSetup(controller as IosWebViewController);
        } else if (Platform.isAndroid) {
          additionalAndroidSetup(controller as AndroidWebViewController);
        }
      },
    );
  }
}
```

However, this can only be done after the iOS platform implementation is transitioned to pigeon and we update the platform interface. After https://github.com/flutter/plugins/pull/4503 lands, this should be possible for Android. So we may want to hold off on adding this feature until iOS is also transitioned.

Reply:
Not sure if I'm missing something with your example @bparrishMines, but `limitsNavigationsToAppBoundDomains` is not a setting that can be changed after the `WebView` has been initialized. It needs to be configured during the initialize process of the `FLTWebViewController` object. And if I understand it correctly `onWebViewCreated` is called after both the WebView Configuration and the WebView is done initializing

Reply:
Ah, I hadn't realized it was part of initialization. Then, I agree with @stuartmorgan that we could expose a `CreationParams` class that each platform can add additional fields too. A solution that could look something like this in the platform interface: 

```dart
class CreationParams {
  factory CreationParams() {
    return WebView.platform.createCreationParams();
  }
}

abstract class WebViewPlatform {
.
.
.
  void createCreationParams() {
    throw UnimplementedError();
  }
}
```

@stuartmorgan Is this what you were considering?

Reply:
That looks pretty much like I was thinking (except that we'd probably want the platform interface class to return the base class rather than throw unimplemented, so that only platforms that want extra parameters need to implement it).

Reply:
@stuartmorgan is this something I should implement, or should I wait for it to be implemented before adding my changes? Also, how would this look in an app where I might target multiple platforms and want toggle options for multiple platforms?

Reply:
It's not something there's an existing plan to implement, so you should go ahead and implement it. I would start with trying it as part of this PR; we could do it as a separate PR that lands first, but it's probably useful to build (and review) in conjunction with this use as a test case to make sure it's working as we want it to end to end. (If it turns out to be more complex than anticipated for some reason, it could be split out later.)

> Also, how would this look in an app where I might target multiple platforms and want toggle options for multiple platforms?

Basically yike [the example shown above](https://github.com/flutter/plugins/pull/4339#discussion_r749486889), but with the settings object being cast, not the controller.

Reply:
> so you should go ahead and implement it

Oh, but as @bparrishMines mentioned, it should wait a couple of weeks since the restructuring that's happening in the webview implementation will probably interact with this.

Reply:
@stuartmorgan Sounds good, is there an active issue or PR for the iOS transition to pigeon that I can follow to know when this change is landed?

Reply:
There wasn't, but really should be. I filed https://github.com/flutter/flutter/issues/93732 (Android is in progress now and nearing completion, iOS will follow after).